core/src/test/java/org/elasticsearch/action/admin/indices/TransportAnalyzeActionTests.java

```diff
@@ -332,10 +337,60 @@ public void testNormalizerWithIndex() throws IOException {
         AnalyzeRequest request = new AnalyzeRequest("index");
         request.normalizer("my_normalizer");
         request.text("ABc");
-        AnalyzeResponse analyze = TransportAnalyzeAction.analyze(request, "text", null, indexAnalyzers, registry, environment);
+        AnalyzeResponse analyze = TransportAnalyzeAction.analyze(request, "text", null, indexAnalyzers, registry, environment, maxTokenCount);
         List<AnalyzeResponse.AnalyzeToken> tokens = analyze.getTokens();
 
         assertEquals(1, tokens.size());
         assertEquals("abc", tokens.get(0).getTerm());
     }
+
+    public void testExceedDefaultMaxTokenLimit() throws IOException{
+        // create a string with No. words more than maxTokenCount
+        StringBuilder sbText = new StringBuilder();
+        for (int i = 0; i <= maxTokenCount; i++){
+            sbText.append('a');
+            sbText.append(' ');
+        }
+        String text = sbText.toString();
+
+        // request with explain=false to test simpleAnalyze path in TransportAnalyzeAction
+        AnalyzeRequest request = new AnalyzeRequest();
+        request.text(text);
+        request.analyzer("standard");
+        IllegalStateException e = expectThrows(IllegalStateException.class,
+            () -> TransportAnalyzeAction.analyze(
+                request, "text", null, null, registry, environment, maxTokenCount));
+        assertEquals(e.getMessage(), "The number of tokens produced by calling _analyze has exceeded the allowed maximum of ["
+            + maxTokenCount + "]." + " This limit can be set by changing the [index.analyze.max_token_count] index level setting.");
+
+        // request with explain=true to test detailAnalyze path in TransportAnalyzeAction
+        AnalyzeRequest request2 = new AnalyzeRequest();
+        request2.text(text);
+        request2.analyzer("standard");
+        request2.explain(true);
+        IllegalStateException e2 = expectThrows(IllegalStateException.class,
+            () -> TransportAnalyzeAction.analyze(
+                request2, "text", null, null, registry, environment, maxTokenCount));
+        assertEquals(e2.getMessage(), "The number of tokens produced by calling _analyze has exceeded the allowed maximum of ["
+            + maxTokenCount + "]." + " This limit can be set by changing the [index.analyze.max_token_count] index level setting.");
+    }
+
+    public void testExceedSetMaxTokenLimit() throws IOException{
```

Comment:
I think this is the case thats equivalent to calling the "_analysis" endpoint on an explicit index. If so, please add this as a small comment to the test method.